### DISCUSSION TITLE
HTTPS issue

### DISCUSSION THREAD

**Discussion by ltsz16** [justcoding121/Titanium-Web-Proxy#910] [Q&A] [answered]
Might be missing something in the setup but when people go to any HTTPS page via the proxy the page comes back without formatting.  Is there something we need to do with a cert. setup or proxy setup so that https pages pull across the wire correctly?

**Comment by justcoding121** [comment, marked as answer]
Sounds like CSS/JS aren't loading ? usually a cert trust issue, not "missing formatting" from the HTML itself.

Checklist:
1. Create + trust the root cert on the *client* machine:
```csharp
proxyServer.CertificateManager.CreateRootCertificate();
proxyServer.CertificateManager.TrustRootCertificate(); // or TrustRootCertificateAsAdmin()
```
2. Confirm `ExplicitProxyEndPoint(..., decryptSsl: true)` and that the browser trusts "Titanium Root Certificate Authority".
3. In DevTools ? Network, check if stylesheet/script requests are failing (often `NET::ERR_CERT_*`).

If you're rewriting HTML in `BeforeResponse`, make sure you're not breaking relative URLs or stripping link tags.

Closing for now ? reopen with a failing URL + any console/cert errors if it's still broken after trusting the root.
